app/components/scheduled_post_indicator/scheduled_post_indicator.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Text, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import FormattedTime from '@components/formatted_time';
+import {useTheme} from '@context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {getUserTimezone} from '@utils/user';
+
+import type UserModel from '@typings/database/models/servers/user';
+
+const WRAPPER_HEIGHT = 40;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        wrapper: {
+            height: WRAPPER_HEIGHT,
+        },
+        container: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            paddingVertical: 12,
+            paddingBottom: 20,
+            paddingHorizontal: 16,
+            color: changeOpacity(theme.centerChannelColor, 0.72),
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+            gap: 12,
+            position: 'absolute',
+            top: 0,
+            width: '100%',
+        },
+        text: {
+            color: changeOpacity(theme.centerChannelColor, 0.75),
+            fontSize: 14,
+        },
+        link: {
+            color: theme.linkColor,
+            fontSize: 14,
+        },
+    };
+});
+
+type Props = {
+    currentUser?: UserModel;
+    isMilitaryTime: boolean;
+    isThread?: boolean;
+    scheduledPostCount?: number;
+}
+
+export function ScheduledPostIndicator({currentUser, isMilitaryTime, isThread, scheduledPostCount = 0}: Props) {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    let scheduledPostText: React.ReactNode;
+
+    if (scheduledPostCount <= 0) {
+        return null;
+    } else if (scheduledPostCount === 1) {
```

Comment:
What happens if it's more than 1?  You will not show dateTime at all?

Why not?

Reply:
Correct, because we cannot show date and time of all scheduled posts in a small indicator in channel. See the attached screenshot in PR description for example.